android/modules/ui/src/java/ti/modules/titanium/ui/PickerProxy.java

```diff
@@ -778,4 +780,45 @@ public String getApiName()
 	{
 		return "Ti.UI.Picker";
 	}
+
+	@Kroll.getProperty
+	public Date getMinDate()
+	{
+		return returnMinMaxDate(TiC.PROPERTY_MIN_DATE);
+	}
+
+	@Kroll.getProperty
+	public Date getMaxDate()
+	{
+		return returnMinMaxDate(TiC.PROPERTY_MAX_DATE);
+	}
+
+	private Date returnMinMaxDate(String dateType)
+	{
+		if (type == UIModule.PICKER_TYPE_DATE) {
+			DatePicker datePicker = ((DatePicker) this.getOrCreateView().getNativeView());
+			if (datePicker != null) {
+				return new Date(dateType.equals(TiC.PROPERTY_MIN_DATE) ? datePicker.getMinDate()
+																	   : datePicker.getMaxDate());
+			}
+		}
+		return null;
+	}
+
+	@Kroll.getProperty
+	public Date getValue()
+	{
+		if (type == UIModule.PICKER_TYPE_DATE) {
```

Comment:
This doesn't account for Time pickers. I know Android doesn't support Date + Time pickers for now, but for a Time picker, I think this should likely return the Date object equivalent to "today" at the selected time of the picker.

Reply:
Also, not the point of this PR, but why *don't* we have a Date + Time picker on Android? I mean we support each individually. Why can't we just create a component that includes both (and merges the date and time value into a single Date object?). Maybe this is a good job to do at the JS level, overriding the createPicker method to handle that specific case and have it silently delegate to both pickers under the hood?

Reply:
I agree that we should our own custom Date+Time picker.
I've written a feature request for it here: [TIMOB-27840](https://jira.appcelerator.org/browse/TIMOB-27840)